cmd/scaffold.go

```diff
@@ -482,7 +482,10 @@ func (fnb *FlowNodeBuilder) initMetrics() {
 
 	fnb.Tracer = trace.NewNoopTracer()
 	if fnb.BaseConfig.tracerEnabled {
-		serviceName := fnb.BaseConfig.NodeRole + "-" + fnb.BaseConfig.nodeIDHex[:8]
+		serviceName := fnb.BaseConfig.NodeRole + "-" + fnb.BaseConfig.nodeIDHex[:7]
+		if len(fnb.BaseConfig.nodeIDHex) > 7 {
+			serviceName = fnb.BaseConfig.NodeRole + "-" + fnb.BaseConfig.nodeIDHex[:8]
+		}
```

Comment:
There was a node startup crash without chis change on unstaked nodes with your latest execution updates.
What happened is that some nodes came with a hex of :7, so and indexing on :8 crashed the startup logic.

Reply:
I tried to make sure that the change is still safe. Legacy 8 character IDs will still use their long name, new 7 character ones will use their entire length but node ids smaller or bigger than 7 and 8 will stick to their original behavior meaning small node ids will crash and big ones will trim to 8.

Reply:
FYI I was able to reproduce this on origin/master running localnet with a single observer:
```
panic: runtime error: slice bounds out of range [:8] with length 7

goroutine 1 [running]:
github.com/onflow/flow-go/cmd.(*FlowNodeBuilder).initMetrics(0xc000527810)
        /app/cmd/scaffold.go:485 +0x61f
github.com/onflow/flow-go/cmd.(*FlowNodeBuilder).onStart(0xc000527810)
        /app/cmd/scaffold.go:1294 +0xf0
github.com/onflow/flow-go/cmd.(*FlowNodeBuilder).Build(0xc000527810)
        /app/cmd/scaffold.go:1264 +0x33
github.com/onflow/flow-go/cmd/access/node_builder.(*UnstakedAccessNodeBuilder).Build(0xc00000f788)
        /app/cmd/access/node_builder/unstaked_access_node_builder.go:272 +0x4f
main.main()
        /app/cmd/access/main.go:29 +0x17e
root@flow-0603:~/stable/work/integration/localnet# git status
On branch master
Your branch is up to date with 'origin/master'.
```

Reply:
You shouldn't need to check the length if you fetch the nodeID that way. If the value is less than 8 bytes it's a bug. 

Originally, it was failing because `fnb.BaseConfig.nodeIDHex` is set based on the CLI flag, which we don't use for observers, so the value was the string `not set`. The fix is to use the resolved NodeID which must be set for all nodes and observers, and must be 32 bytes